pkg/synthetictests/historicaldata/alert_types.go

```diff
@@ -0,0 +1,141 @@
+package historicaldata
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/openshift/origin/pkg/synthetictests/platformidentification"
+)
+
+type AlertStatisticalData struct {
+	AlertDataKey `json:",inline"`
+	Name         string
+	P95          float64
+	P99          float64
+	JobRuns      int64
+}
+
+type AlertDataKey struct {
+	AlertName      string
+	AlertNamespace string
+	AlertLevel     string
+
+	platformidentification.JobType `json:",inline"`
+}
+
+type AlertBestMatcher struct {
+	HistoricalData map[AlertDataKey]AlertStatisticalData
+}
```

Comment:
I tried multiple times to re-use the old system where both alerts and disruption were mapping to the same Go types, but I couldn't pull it off. After About 3 tries I gave up and forked the code. The BestMatchers remain using the common code, which I felt was the most important to keep shared between the two.

Reply:
where is the "common code" that you refer to?  alert_types.go and disruption_types.go look almost identical (except for the types and names).

Reply:
ah.  I'm glad you didn't use generics for now.